packages/opentelemetry-sdk-setup/src/sdk-setup.ts

```diff
@@ -0,0 +1,104 @@
+  preloadingEnabled: boolean
+  httpProtocol: string
+  host: string
+  port: number
+  /** API Key used for a dedicated trace provider */
+  apiKey: string
+  /** Sample rate being used for this trace, this allows for consistent probability sampling */
+  sampleRate: number
+  /** Properties of the root span and trace id used to stitch context */
+  traceId?: string
+  traceFlags?: number
+  parentSpanId?: string
+  baggageFilePath?: string
+  /** Debug mode enabled - logs to stdout */
+  debug: boolean
+  /** System log file descriptor */
+  systemLogFile?: number
+}
+
+let sdk: HoneycombSDK | undefined
+
+/** Starts the tracing SDK, if there's already a tracing service this will be a no-op */
+export const startTracing = async function (options: TracingOptions, packageJson: PackageJson) {
+  if (!options.preloadingEnabled) return
+  if (sdk) return
+
+  sdk = new HoneycombSDK({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_VERSION]: packageJson.version,
+    }),
+    serviceName: packageJson.name,
+    protocol: 'grpc',
+    apiKey: options.apiKey,
+    endpoint: `${options.httpProtocol}://${options.host}:${options.port}`,
+    sampleRate: options.sampleRate,
+    // Turn off auto resource detection so that we fully control the attributes we export
+    autoDetectResources: false,
+  })
+
+  // Set the diagnostics logger to our system logger. We also need to suppress the override msg
+  // in case there's a default console logger already registered (it would log a msg to it)
+  diag.setLogger(getOtelLogger(options.debug, options.systemLogFile), {
+    logLevel: options.debug ? DiagLogLevel.DEBUG : DiagLogLevel.INFO,
+    suppressOverrideMessage: true,
+  })
+
+  sdk.start()
+
+  // Loads the contents of the passed baggageFilePath into the baggage
+  const baggageAttributes = await loadBaggageFromFile(options.baggageFilePath)
+  const baggageCtx = setMultiSpanAttributes(baggageAttributes)
+
+  const traceFlags = options.traceFlags !== undefined ? options.traceFlags : TraceFlags.NONE
+  // Sets the current trace ID and span ID based on the options received
+  // this is used as a way to propagate trace context from other processes such as Buildbot
+  if (options.traceId !== undefined && options.parentSpanId !== undefined) {
+    return trace.setSpanContext(baggageCtx, {
+      traceId: options.traceId,
+      spanId: options.parentSpanId,
+      traceFlags: traceFlags,
+      isRemote: true,
+    })
+  }
+
+  return context.active()
+}
+
+/** Stops the tracing service if there's one running. This will flush any ongoing events */
+export const stopTracing = async function () {
+  if (!sdk) return
+  try {
+    // The shutdown method might return an error if we fail to flush the traces
+    // We handle it and use our diagnostics logger
+    await sdk.shutdown()
+    sdk = undefined
+  } catch (e) {
+    diag.error(e)
+  }
+}
```

Comment:
This whole start/stop logic is a move of:
- https://github.com/netlify/build/blob/97fd98c98c1a6d0ecbca8a5e5ca14867d3e6053e/packages/build/src/tracing/main.ts#L32-L79

Once we fully roll this out in prod we can delete and remove build's sdk setup.